core/views.py

```diff
@@ -13,6 +14,8 @@ def base_css(request):
 @csrf_exempt
 def login(request):
     next = request.GET.get('next') or request.POST.get('next') or '/'
+    if not url_has_allowed_host_and_scheme(next, settings.ALLOWED_HOSTS):
```

Comment:
Per the Django code, you'll also need to ensure you're encoding the path component of the redirect URL with `django.utils.encoding.iri_to_uri()`: https://github.com/django/django/blob/d3d173425fc0a1107836da5b4567f1c88253191b/django/utils/http.py#L253

Reply:
Thanks a lot @MarkusH, you're completely right :)